src/mc-fle2-payload-iev-v2.c

```diff
@@ -429,9 +483,55 @@ bool mc_FLE2IndexedRangeEncryptedValueV2_parse(mc_FLE2IndexedEncryptedValueV2_t
     const uint64_t SEV_len = SEV_and_edges_len - edges_len;
     CHECK_AND_RETURN(mc_reader_read_buffer(&reader, &iev->ServerEncryptedValue, SEV_len, status));
 
-    // Ignore Metadata block.
-    BSON_ASSERT(mc_reader_get_remaining_length(&reader) == edges_len);
+    iev->metadata = (_mongocrypt_buffer_t *)malloc(iev->edge_count * sizeof(_mongocrypt_buffer_t));
```

Comment:
```suggestion
    iev->metadata = (_mongocrypt_buffer_t *)bson_malloc0(iev->edge_count * sizeof(_mongocrypt_buffer_t));
```

`bson_` because it keeps it in the correct allocator buckets.
`0` so that it zeroes out the memory it's allocated (similar to `calloc`)